solutions/cses-1139.mdx

```diff
@@ -0,0 +1,124 @@
+---
+id: cses-1139
+source: CSES
+title: Distinct Colors
+author: Timothy Gao
+---
+
+## Editorial
+
+Let us consider the euler tour of the tree. We flatten the tree into an arary, where each node corresponds to a range 
+in this array. Now, we are essentially tasked the find the number of distinct values in a range $N$ times(for each node).
+
+There are a variety of ways to approach the problem from here.
+
+**Solution 1: PURS**
+
+One such solution is with a PURS structure. Let us consider iterating through the euler tour array from left to right. 
+When we consider a node at index i in the euler tour array, it will have it's subtree range be $[j, i]$, where $j<=i$.
+Now, let's focus on a single color. Notice that if there are multiple of color $c$ to the left of a certain index $i$,
+we really only care about $c$'s most recent(rightmost) occurance. This is because any none-rightmost occurance of $c$
+from $i$ must include the right most occurance of $c$. If that's confusing to you, we're essentially saying that 
+any segment $[l, r]$ must contains $i$ if $l <= i <= r$. 
+
+With this observation, we can essentially reduce the distinct colors queries to a simple range query. As we iterate
+ through the euler tour from left to right, mark the current index as 1, and if the color of the node at current index 
```

Comment:
```suggestion
through the euler tour from left to right, mark the current index as 1, and if the color of the node at the current index 
```
When you said "mark the current index as 1" are you sure you didn't mean "mark the current index as $i$" (I didn't actually read the problem :laughing: so idk).